### PR TITLE
Update makefile

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

**Comment by kyessenov** [comment]
@wora I've added google/type, google/api/annotations.proto, and google/api/http.proto. That should cover most common google protos.

File: Makefile [hunk 1]
```diff
@@ -1,55 +1,82 @@
...
-	# GOOGLEAPIS Generation
-	@mkdir -p ${GOOGLEAPIS}/$(RPC_PATH)
+gogoslick: depend
+	@go build -o gogoslick vendor/github.com/gogo/protobuf/protoc-gen-gogoslick/main.go
 
-	## Generate google/rpc/status.pb.go
-	@curl -sS $(GOOGLEAPIS_URL)/google/rpc/status.proto -o $(STATUS_PROTO)
-	@gogoreplace 'option go_package = "google.golang.org/genproto/googleapis/rpc/status;status";' '' $(STATUS_PROTO)
-	@$(PROTOC) $(PROTO_PATH) $(PLUGIN) $(STATUS_PROTO)
+$(googleapis_protos): %:
```
Comment (ldemailly) [inline Makefile]: nice !

File: Makefile [hunk 2]
```diff
@@ -1,55 +1,82 @@
...
+	google/protobuf/wrappers.proto=$(GOGO_TYPES) \
 
-PROTOC = protoc-min-version -version=3.0.0
+comma := ,
+empty :=
+space := $(empty) $(empty)
+mapping_with_spaces := $(foreach map,$(importmaps),M$(map),)
+MAPPING := $(subst $(space),$(empty),$(mapping_with_spaces))
+PLUGIN := --plugin=protoc-gen-gogoslick=gogoslick --gogoslick_out=$(MAPPING):googleapis
+PROTOC = protoc
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply Makefile]: I didn't see any build set-up here, and the doc clearly states the version.
Should we commit a Dockerfile instead?

File: Makefile [hunk 3]
```diff
@@ -1,55 +1,82 @@
...
+	$(foreach var,$(googleapis_packages),mkdir -p googleapis/$(var);)
+
+protoc.version:
 	# Record protoc version
 	@echo `protoc --version` > protoc.version
 
-	# GOOGLEAPIS Generation
-	@mkdir -p ${GOOGLEAPIS}/$(RPC_PATH)
+gogoslick: depend
+	@go build -o gogoslick vendor/github.com/gogo/protobuf/protoc-gen-gogoslick/main.go
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply Makefile]: What do you mean? dep checks out gogo at version 0.5.

File: Makefile [hunk 4]
```diff
@@ -1,55 +1,82 @@
...
+	google/protobuf/timestamp.proto=$(GOGO_TYPES) \
+	google/protobuf/wrappers.proto=$(GOGO_TYPES) \
 
-PROTOC = protoc-min-version -version=3.0.0
+comma := ,
+empty :=
+space := $(empty) $(empty)
+mapping_with_spaces := $(foreach map,$(importmaps),M$(map),)
+MAPPING := $(subst $(space),$(empty),$(mapping_with_spaces))
+PLUGIN := --plugin=protoc-gen-gogoslick=gogoslick --gogoslick_out=$(MAPPING):googleapis
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply Makefile]: This is to override PATH-based resolution with an exact binary name.

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (douglas-reid) [reply Makefile]: elsewhere, we build `-o gogoslick-$(GOGO_VERSION)`, based on feedback from PRs. I don't feel all that strongly, just wondering if we should be consistent.

**Pull request merged**